uxmhf/uobjcoll/uobjs/main/uobjs/debug/dbgapi.c

```diff
@@ -1,172 +1,197 @@
-
-  return;
-}
-
-
-
-void uberspark_uobjrtl_debug__init(char *params){
-  xmhfhw_platform_serial_init(params);
-}
-
-void dbgprintf (const char *fmt, ...){
-    va_list       ap;
-	int retval;
-	char buffer[1024];
-
-	va_start(ap, fmt);
-	retval = vsnprintf((char *)&buffer, 1024, (const char *)fmt, ap);
-	uberspark_uobjrtl_hw__generic_x86_32_intel__spin_lock(&libxmhfdebug_lock);
-	xmhfhw_platform_serial_puts((char *)&buffer);
-	uberspark_uobjrtl_hw__generic_x86_32_intel__spin_unlock(&libxmhfdebug_lock);
-    va_end(ap);
-}
-
-#else
-
-void uberspark_uobjrtl_debug__init(char *params){
-	(void)params;
-}
-
-
-#endif // defined
+/*-
+ * Copyright (c) 1986, 1988, 1991, 1993
+ *	The Regents of the University of California.  All rights reserved.
+ * (c) UNIX System Laboratories, Inc.
+ * All or some portions of this file are derived from material licensed
+ * to the University of California by American Telephone and Telegraph
+ * Co. or Unix System Laboratories, Inc. and are reproduced herein with
+ * the permission of UNIX System Laboratories, Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 4. Neither the name of the University nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ *
+ *	@(#)subr_prf.c	8.3 (Berkeley) 1/21/94
+ */
+
+/*
+ * debug api
+ *
+ * author: amit vasudevan (amitvasudevan@acm.org)
+ */
+
+#include <uberspark/uobjcoll/platform/pc/uxmhf/uobjs/main/include/xmhf.h>
+
+#if defined (__UBERSPARK_UOBJCOLL_CONFIGDEF_DEBUG_MEMORY__)
+
+// Memory storing uberXMHF output. 1MB in size.
+#define _LOG_MEM_SZ   (1*1024*1024)
+__attribute__((aligned(4096))) static char _log_mem[_LOG_MEM_SZ];
```

Comment:
Should we make this a global variable (drop the static) so that we can use something like `readelf` on the uxmhf elf binary to find the address of _log_mem?